### ISSUE TITLE
independent SNP and unfolded SFS

### ISSUE THREAD

**Issue by biozzq** [isaacovercast/easySFS#2]
Hi @isaacovercast 

Very nice tools. After reading README.md, I can not understand `independent SNP` well, does it mean I should prune my SNPs data (my SNPs come from whole genome re-sequencing)? In addition, if I want to generate unfolded SFS when I know the ancestral state, can i use `easySFS`?

Best

**Comment by Guillefriis** [comment]
Hello, by polarized vcf you mean to have used an ancestral sequence as a reference for the variant calling process? Thank you

**Comment by isaacovercast** [comment]
Yes, that's correct.

**Comment by Guillefriis** [comment]
Thanks for such a quick response Isaac. Since this thread is still active, I have one more question: I'm using easySFS to produce fastsimcoal2 input files, and it's necessary to set the number of samples for each analyzed population. My understanding is that I should provide the number used in the projection with easySFS, rather than the original number of samples of my dataset, is that correct?

**Comment by isaacovercast** [comment]
Yes, choose the projection value that maximizes the number of segregating sites for your populations. This is the current best practice.

**Comment by Guillefriis** [comment]
Great, appreciate it.

**Comment by Guillefriis** [comment]
Hello Isaac, I'm using easySFS in a new dataset, I need to produce 100 bootstrapped SFS from a vcf file. I can easily write a script bootstrapping my vcf file and produce the SFS with your program, but given how the downsampling works, I would end with SFS based in different samples/positions! I reckon I could use the datadict.txt file to produce a vcf file corresponding to the original variants used in the projection, and bootstrap that one, but still don't quite understand how, is it doable? Many thanks!

**Comment by isaacovercast** [comment]
@Guillefriis Hello. I'm not sure I understand the problem exactly. The downsampling is deterministic, so there wouldn't be any variation in the samples for each bootstrap, and it would resample one snp per locus (or genomic window), which it seems like is what you want. Maybe I don't understand well what you are trying to do, so let me know about that.

**Comment by Guillefriis** [comment]
Hi @isaacovercast , thanks for such I quick response. Maybe I don't quite well understand how the projection works, it may be easier if I explain my case indeed: I have a vcf file with 26 diploid individuals from 2 populations and ~ 700K SNPs. I selected --proj=16,16 based on the --preview results, amounting to 71K and 86K variable positions. I can now generate 100 bootstrapped vcf and run easySFS in loop for each one with --proj=16,16. But is this correct? The bootstrapped vcf's are (to some extent) different from the original, so variant positions accounting individuals not previously used in the original projection may end  up been included in the bootstrapped SFS, instead of just resampled positions with replacement of those used in the original projection, am I right? Can this be solved generating a subset of the original vcf file with only the projected variants and individuals in first place? I hope this is clear enough, thanks!

**Comment by isaacovercast** [comment]
@Guillefriis I guess I still don't understand the problem. When you perform the down projection, the notion of "individuals" no longer applies. It's not sampling 16 random individuals, it's averaging over snp frequencies for all possible combinations of samples of 16 individuals. The "bootstrapped vcfs" are not different at all, the resampling of snps will happen as you want, without a problem.

**Comment by Guillefriis** [comment]
Huh ok thanks Isaac, I guess I need to look more into it and a clearer idea of how it works, cheers!
